samples/HealthChecksSample/GCInfoHealthCheck.cs

```diff
@@ -2,21 +2,53 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
 
 namespace HealthChecksSample
 {
     // This is an example of a custom health check that implements IHealthCheck.
     //
-    // See CustomWriterStartup to see how this is registered.
+    // This example also shows a technique for authoring a health check that needs to be registered
+    // with additional configuration data. This technique works via named options, and is useful
+    // for authoring health checks that can be disctributed as libraries.
+    
+    public static class GCInfoHealthCheckBuilderExtensions
```

Comment:
This is a sample that hews pretty close to what we'd tell a library author to do if they want to ship a health check as a package.